File: PZ-Switch/src/zombie/PredicatedFileWatcher.cpp
#include "zombie/PredicatedFileWatcher.h"

namespace zombie {

public
PredicatedFileWatcher::PredicatedFileWatcher(
 Predicate<String> predicate,
 PredicatedFileWatcher.IPredicatedFileWatcherCallback callback) {
 // TODO: Implement PredicatedFileWatcher
 return nullptr;
}

public
PredicatedFileWatcher::PredicatedFileWatcher(
 const std::string &path,
 PredicatedFileWatcher.IPredicatedFileWatcherCallback callback) {
 // TODO: Implement PredicatedFileWatcher
 return nullptr;
}

public
PredicatedFileWatcher::PredicatedFileWatcher(
 const std::string &path, Predicate<String> predicate,
 PredicatedFileWatcher.IPredicatedFileWatcherCallback callback) {
 // TODO: Implement PredicatedFileWatcher
 return nullptr;
}

std::string PredicatedFileWatcher::getPath() {
 // TODO: Implement getPath
 return "";
}

std::string PredicatedFileWatcher::processPath(const std::string &string) {
 // TODO: Implement processPath
 return "";
}

bool PredicatedFileWatcher::pathsEqual(const std::string &string) {
 // TODO: Implement pathsEqual
 return false;
}

void PredicatedFileWatcher::onModified(const std::string &entryKey) {
 // TODO: Implement onModified
}

void PredicatedFileWatcher::call(const std::string &string) {
 // TODO: Implement call
}

void PredicatedFileWatcher::call(T var1) {
 // TODO: Implement call
}

void PredicatedFileWatcher::call(const std::string &entryKey) {
 // TODO: Implement call
}

} // namespace zombie

File: converted/source2/source/zombie/PredicatedFileWatcher.java
package zombie;

import java.io.File;
import java.util.function.Predicate;
import zombie.PredicatedFileWatcher.GenericPredicatedFileWatcherCallback;
import zombie.PredicatedFileWatcher.IPredicatedDataPacketFileWatcherCallback;
import zombie.PredicatedFileWatcher.IPredicatedFileWatcherCallback;

public final class PredicatedFileWatcher {
   private final String m_path;
   private final Predicate<String> m_predicate;
   private final IPredicatedFileWatcherCallback m_callback;

   public PredicatedFileWatcher(Predicate<String> var1, IPredicatedFileWatcherCallback var2) {
      this(null, var1, var2);
   }

   public PredicatedFileWatcher(String var1, IPredicatedFileWatcherCallback var2) {
      this(var1, null, var2);
   }

   public <T> PredicatedFileWatcher(String var1, Class<T> var2, IPredicatedDataPacketFileWatcherCallback<T> var3) {
      this(var1, null, new GenericPredicatedFileWatcherCallback(var2, var3));
   }

   public PredicatedFileWatcher(String var1, Predicate<String> var2, IPredicatedFileWatcherCallback var3) {
      this.m_path = this.processPath(var1);
      this.m_predicate = var2 != null ? var2 : this::pathsEqual;
      this.m_callback = var3;
   }

   public String getPath() {
      return this.m_path;
   }

   private String processPath(String var1) {
      return var1 != null ? ZomboidFileSystem.processFilePath(var1, File.separatorChar) : null;
   }

   private boolean pathsEqual(String var1) {
      return var1.equals(this.m_path);
   }

   public void onModified(String var1) {
      if (this.m_predicate.test(var1)) {
         this.m_callback.call(var1);
      }
   }
}
